clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -1710,19 +1724,18 @@ public DeleteTopicsResult deleteTopics(final Collection<String> topicNames,
                 Collections.emptyMap(), now, deadline);
             runnable.call(call, now);
         }
-        return new DeleteTopicsResult(new HashMap<>(topicFutures));
+        return new HashMap<>(topicFutures);
     }
 
-    @Override
-    public DeleteTopicsWithIdsResult deleteTopicsWithIds(final Collection<Uuid> topicIds,
-                                                         final DeleteTopicsOptions options) {
+    private Map<Uuid, KafkaFuture<Void>> handleDeleteTopicsUsingIds(final Collection<Uuid> topicIds,
+                                                                    final DeleteTopicsOptions options) {
         final Map<Uuid, KafkaFutureImpl<Void>> topicFutures = new HashMap<>(topicIds.size());
         final List<Uuid> validTopicIds = new ArrayList<>(topicIds.size());
         for (Uuid topicId : topicIds) {
             if (topicId.equals(Uuid.ZERO_UUID)) {
                 KafkaFutureImpl<Void> future = new KafkaFutureImpl<>();
                 future.completeExceptionally(new UnknownTopicIdException("The given topic ID '" +
-                        topicId + "' cannot be represented in a request."));
+                    topicId + "' cannot be represented in a request."));
```

Comment:
This message is a little strange. We can certainly represent the topic id, but it is invalid. I wonder if it would make sense to raise `IllegalArgumentException` directly instead of through the result since this is likely a logical error of some kind.

Reply:
Ah yeah. The name equivalent is `InvalidTopicException`  so maybe we could just use that (or `IllegalArugmentException` is probably better) so the cases are similar? I think that we should keep it as through the result so it is consistent with the name equivalent, but that's just me.

Reply:
Hmm, I like `InvalidTopicException` over `IllegalArgumentException` if we are raising it through the future. Typically `IllegalArgumentException` is raised directly.

Reply:
Ok. We can change to that.